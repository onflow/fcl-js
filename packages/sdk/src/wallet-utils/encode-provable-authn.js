import {withPrefix} from "@onflow/util-address"
import {invariant} from "@onflow/util-invariant"
import {encode} from '@onflow/rlp';

export const encodeMessageFromProvableAuthn = (address, timestamp, appTag = "") => {
    invariant(address, "Encode Message From Provable Authn Error: address must be defined")
    invariant(timestamp, "Encode Message From Provable Authn Error: timestamp must be defined")

    return encode([appTag, withPrefix(address), timestamp]).toString("hex")
}
