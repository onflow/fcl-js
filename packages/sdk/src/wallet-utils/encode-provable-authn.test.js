import {encodeMessageFromProvableAuthn} from "./encode-provable-authn.js"

const address = "0xABC123DEF456"
const timestamp = 1632179933495
const appTag = "FLOW-JS-SDK"

describe("encode provable authn", () => {
    test("encode provable authn with appTag", () => {
        const message = encodeMessageFromProvableAuthn(address, timestamp, appTag)

        expect(message).toMatchSnapshot()
    })

    test("encode provable authn without appTag", () => {
        const message = encodeMessageFromProvableAuthn(address, timestamp)

        expect(message).toMatchSnapshot()
    })
})
