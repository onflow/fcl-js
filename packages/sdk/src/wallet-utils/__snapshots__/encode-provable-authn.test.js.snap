// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encode provable authn encode provable authn with appTag 1`] = `"da8b464c4f572d4a532d53444b86abc123def45686017c05815137"`;

exports[`encode provable authn encode provable authn without appTag 1`] = `"cf8086abc123def45686017c05815137"`;
